components/rich_text_editor/rich_text_editor.test.js

```diff
@@ -0,0 +1,197 @@
+import sinon from 'sinon';
+import { assert } from 'chai';
+import { mount, createLocalVue } from '@vue/test-utils';
+import DtRichTextEditor from './rich_text_editor.vue';
+import { EditorContent } from '@tiptap/vue-2';
+
+// Wrappers
+let wrapper;
+let editor;
+
+// Test Environment
+let propsData;
+let attrs;
+let slots;
+let listeners;
+let inputStub;
+const localVue = createLocalVue();
+
+// Constants
+const baseProps = {
+  value: 'initial value',
+  inputAriaLabel: 'aria-label text',
+};
+
+// Helpers
+const _setChildWrappers = () => {
+  editor = wrapper.find('[data-qa="dt-rich-text-editor"]').find('div[contenteditable]');
+};
+
+const _mountWrapper = () => {
+  wrapper = mount(DtRichTextEditor, {
+    propsData,
+    components: { EditorContent },
+    listeners,
+    attrs,
+    slots,
+    localVue,
+  });
+};
+
+async function mountWithProps (newProps) {
+  // Can't use setProps since the Editor instance is created in mounted().
+  // There are helper methods we could use to update the editor options
+  // (editor.setOptions), but we want to test that the passed props are
+  // properly getting through to the editor, so rather re-mount.
```

Comment:
My only issue with this is that it will not capture any reactivity problems. It's a pretty common problem (especially when using 3rd party libraries) that a component will mount successfully with the passed in props, but then doesn't react when props are changed after it's rendered. 

It's fine to do it this way for the initial mount but we should maybe also test props changes with setProps after rendering the component because otherwise all your "watch" code is currently not covered by tests.

Reply:
Yeah good point! I think I wrote these tests before I had that watcher set up haha. And to prove your point, the tests did find a bug right away! The aria-readonly was being set backwards, so I jut slipped that fix in this PR if that's ok.